app/helpers/application_helper.rb

```diff
@@ -206,7 +206,7 @@ def searchable?
 
   def auto_complete_search(name, val, options = {})
     path = options[:path] || send("#{controller_name}_path")
```

Comment:
Or, what about just adding the `auto_complete_search` to the path on pages where it's needed?